### ISSUE TITLE
Newbie question, cannot get quickstart sample working

### ISSUE THREAD

**Issue by daveb49** [olifolkerd/tabulator#763]
Hi
    I'm a newbie and not sure of overall webpage setup. Have been trying variations to get the supplied quickstart sample working and cannot get any table output at all. That is, I don't get to the step 4 (=smile) step.

Is there a *complete* html page quickstart/beginners sample somewhere I can look at?
I've pasted all the steps into place but obviously I'm getting it wrong/missing some setup step.

I'll attach one of my many attempts:
[tabulatortest.html.txt](https://github.com/olifolkerd/tabulator/files/1543678/tabulatortest.html.txt)

[renamed to .txt for upload here, but tried as .html on test server and it displays the text before and after the table but nothing in between]

tks for any help
Dave

PS I am hoping to use tabulator to get and display json data which is obtained from a GET posted to an http url. An example of how to do that would be great

**Comment by daveb49** [comment]
Still struggling... help!!!
I re-read the quickstart line by line and tried a slight tweak but still can't find the magic sauce.
Must be close though surely????
Here is my latest attempt...
[tabulatortest2.html.txt](https://github.com/olifolkerd/tabulator/files/1545356/tabulatortest2.html.txt)

Anyone? anyone?

**Comment by daveb49** [comment]
Aha, more random perturbations of the header and I now have it working.
I do think it would be good to give a complete working example like this in the QuickStart, but in case anyone else needs it they might find this useful:

[tabulatortest3-working.html.txt](https://github.com/olifolkerd/tabulator/files/1545574/tabulatortest3-working.html.txt)

**Comment by olifolkerd** [comment]
Hey @daveb49,

Im not sure i follow you, the quickstart page has a full working example with code, and the examples section is chocked full of working examples, which bit did you think was missing?

Cheers

Oli

**Comment by daveb49** [comment]
I found lots of code snippets but nowhere any end-to-end complete page including the headers. The problem I had was getting the details in the right place and getting the header correct, I didn’t fully understand what had to go where and also had to find CDN sources for the java UI etc. A working full HTML page would have been a huge help... did I miss seeing it somewhere? Please correct me if so by pointing to a URL.

You can see where I was struggling by comparing the three little uploads I did as I progressed. My comment is that a complete working QuickStart example would have saved me a lot of agony... I spent hours trying different things and am still unsure of why the link href is in the header and how to trigger an update of the data from a json http GET which is my next target.

One other thing... possibly should raise as a separate issue... the QuickStart lets you click on column headers to sort by and this is great but the date column doesn’t respond at all, certainly doesn’t sort ... just reverses the little direction arrow indicator. ?? Is this a bug?

Tks 
Dave

PS the reason I persisted at all is that this appears to be a great utility looking at the impressive samples... I don’t think I’ll have any problem using it now I’ve got to first base... thanks!!

**Comment by olifolkerd** [comment]
Hey @daveb49 

Thanks for your feedback,

The example in the [quick-start guide](http://tabulator.info/quickstart/3.3) is a fully working example with all the code necessary to get the table up and running as are all the examples.

Step 1 one of the guide provides a number of different sources that you can get the source files for including the library, It assumes a degree of previous experience in using java script, and most developers will be able to discern from these options how to build the table and choose the appropriate source location for themselves. 

Im afraid that the Tabulator site is there to show developers how to use the library, but not how to build a website from the ground up, if you would like some additional pointers on this i would be happy to point you to a few good sites.

There was an empty date in the list which was making the sort view confusing, i have now removed this to make the example clearer.

Glad to hear you stuck it out.

If you have any other questions about using Tabulator please give me a shout.

Cheers

Oli

**Comment by daveb49** [comment]
Thanks … I certainly agree that I do not have adequate experience with the rules for Jscript.

I did get a lot of help from the W3Schools eg https://www.w3schools.com/js/js_examples.asp

…where all the examples show complete web pages.

The difficulty I had with the QuickStart came down to me not understanding what had to be in the header vs what had to be in the body and not being able to look at any complete working example. Anyhow, moving on…

 

Thanks for addressing the concern about the sorting of dates… but I think the absence of a date when sorting should be treated as another date value and all the empty dates should sort to one end of the list, not disable sorting for the whole column. I would say this is a bug, surely?

 

Many thanks for your responses

Dave

**Comment by daveb49** [comment]
OK, all in motion now. But just repeating this... 

>On the sorting of dates… I think the absence of a date when sorting should be treated as another date value and all the empty dates should sort to one end of the list, not disable sorting for the whole column. I would say this is a bug, surely?

Should I report this separately as a new thread?

**Comment by olifolkerd** [comment]
Hey @daveb49 

Great to hear, at present all sorters only work with data that matches the values they are expected to sort so this is not a bug but.

In the next release the behaviour of sorters is being relaxed to accept  wider range of inputs so it will no longer be a problem.

Cheers

Oli

**Comment by daveb49** [comment]
Sounds good, look fwd to that.

The invalid entries will then accumulate at one end of the sorted column

Rgds Dave

**Comment by E3V3A** [comment]
@daveb49  
I totally agree with you! I had exactly the same problem today, unfortunately I didn't find this post until now! That code should be posted on the main website quick example documentation.

**Comment by olifolkerd** [comment]
Hey @daveb49  and @E3V3A 

As of version 3.4 which will be released next weekend, all sorters will handle null and empty values correctly.

Cheers

Oli :)

**Comment by daveb49** [comment]
Great to hear, Oli. That should make the sorting example "work" by a stricter definition of working where it sorts leaving invalid data at one end of the list.

On E3V3A's comment: I would love to see a refined version of my complete working example included on the QuickStart page... I still don't claim to have lots of expertise in the matter of correct JScript practices in the header setup area, but for what its worth, have eliminated one line from my full working sample... I think the require was redunsant/unnecessary with the tabulator scripts already included from cloudflare... anyhow here's another version that still works...

[tabulatortest4-working.html.txt](https://github.com/olifolkerd/tabulator/files/1697082/tabulatortest4-working.html.txt)

I do feel it would be helpful to others for someone with expertise in this area to review/refine this complete self-contained example and add it or similar to the QuickStart. It's like the difference between explaining the printf statement in C and providing a complete Hello World C program. The QuickStart is great, lots of good material there, but seriously, there is no complete working sample html file.

**Comment by olifolkerd** [comment]
Hey @daveb49,

As i have mentioned before, as with most libraries a basic level of understanding in how to code a website and include a library is expected, and the purpose of the website is not to teach basic HTML and JavaScript.

That being said, i will look at including a link to a rudimentary html page from the quick start link to help beginners out.

Cheers

Oli :)

**Comment by daveb49** [comment]
PS in defence of the need for this "rudimentary html page", I do have decades of programming experience in major programming languages like C, C++ and many others, plus have written many asp and php webpages over the years to get various jobs done. Limited Java,C#,JScript experience. Plus I have a PhD and have run university courses on VLSI hardware, Physics, CompSci/system programming, have written device drivers for Win95 and NT etc etc... yes, indeed, I am a newbie to this precise area, but hopefully represent a large group of people who are essentially competent programmers but will also struggle to get started with the use of this (excellent) Tabulator tool without the help of a complete sample. Looking forward to seeing something added to the QuickStart.

**Comment by olifolkerd** [comment]
@daveb49 I didnt mean that as a personal insult in the slightest.

**Comment by Whiteshua** [comment]
I just want to add to @daveb49's comment about wanting a full rudimentary html page example.  I am in the same boat as far as extensive programming experience, but was not able to get Tabulator working for the same reason.  Only after I searched, found this thread and then CTRL-C- CTRL-V'd his .txt file was I able to get it working.  I do understand your position though @olifolkerd.    Just my two cents

Thanks for this wonderful tool.

**Comment by olifolkerd** [comment]
Hey @Whiteshua 

For the last 6 months The [Quickstart Page](http://tabulator.info/docs/4.1/quickstart) has contained a link to a [Basic HTML Example](http://tabulator.info/basic/4.1) that demonstrates all the basics needed to get things started

Cheers

Oli :)

**Comment by Whiteshua** [comment]
I missed it (obviously).  Sorry about that.  Thanks for the correction.

**Comment by Whiteshua** [comment]
I looked all over your site, including the QuickStart area, but I think I got confused with something about it and couldn’t get it to work, however when I pasted the OPs .txt file in it worked fine.

The funny thing is, now that I look at your QuickStart page again it makes total sense to me and I have no idea how I got so confused.  Lol

I think your page is fine as it is now.  This was my bad.

**Comment by gherson** [comment]
fyi, I had to include jquery and jquery-ui per @daveb49 for the quickstart to appear correctly.  Otherwise the example table appeared mangled.  

I mention this because quickstart does not.  best regards

**Comment by olifolkerd** [comment]
@gherson you only have to do that if you are using the jQuery wrapper which is not covered in the quickstart guide.

Cheers

Oli :)
